README.rst

```diff
@@ -263,7 +268,17 @@ Then update the ``users`` table in the database::
 Synapse Development
 ===================
 
-Join our developer community on Matrix: `#synapse-dev:matrix.org <https://matrix.to/#/#synapse-dev:matrix.org>`_
+There are three great points of reference for developers:
+
+* Our developer community on Matrix: `#synapse-dev:matrix.org <https://matrix.to/#/#synapse-dev:matrix.org>`_,
+* `Documentation specific to development <https://matrix-org.github.io/synapse/latest/development/>`_
```

Comment:
This isn't actually a page at the moment, but just a section that organises other pages. The contributing guide is probably the best entry point for development on Synapse, though I'm aware you link it below already.

Perhaps you may want to first link the contributing guide as a place to start, and then some other bits of the Development section if one wants to go deeper? Such as [how we use git](https://matrix-org.github.io/synapse/latest/development/git.html) and documenting [internal bits of Synapse](https://matrix-org.github.io/synapse/latest/development/internal_documentation/index.html).

Reply:
> Though we may want to hold off until there's actually some version dropdown selection UI first :)

I think readthedocs does this pretty well FWIW, e.g. see the bottom-left of https://example.readthedocs.io/en/latest/